src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -618,12 +618,13 @@ void Transformations::PreLpt(const std::vector<ov::element::Type>& defaultPrecis
     CPU_SET_CALLBACK_COMMON(manager, nmsCallback, ov::pass::ConvertNMS9ToNMSIEInternal);
     CPU_SET_CALLBACK_COMMON(manager, nmsCallback, ov::pass::ConvertMulticlassNmsToMulticlassNmsIE);
     CPU_SET_CALLBACK_COMMON(manager, nmsCallback, ov::pass::ConvertMatrixNmsToMatrixNmsIE);
-    CPU_SET_CALLBACK_COMMON(manager,
-        [this](const_node_ptr &node) -> bool {
+    CPU_SET_CALLBACK_COMMON(
+        manager,
+        [this](const_node_ptr& node) -> bool {
             std::string errorMsg;
-            // Current SDPA impl is optimized only for LLM models, so we decompose it for others to avoid perf regression.
-            // Matching the pattern is a little complicated, so we just check if there is any state nodes.
-            return node::ScaledDotProductAttention::isSupportedOperation(node, errorMsg) && model->get_variables().size() > 0;
+            // Current SDPA impl is optimized only for LLM models, so we decompose it for others to avoid perf
+            // regression. Matching the pattern is a little complicated, so we just check if there is any state nodes.
+            return !node::ScaledDotProductAttention::isSupportedOperation(node, errorMsg);
```

Comment:
If we are seeking for a workaround that can be merged: can we reuse the same logic that detects LoRA patterns used to insert LoRA macro-op here? To trigger the special behavior if the dynamic lora is present and "subtract" these variables from potentially existing LLM kv-cache variables?

Reply:
If it is so performance-sensitive, can we introduce a property for CPU plugin to override this behavior and always decompose or leave SDPA in the model?

Reply:
The problem here is that LLM models may have LoRA adapters too, so the LoRA pattern itself doesn't help to understand whether we have to call the decomposition or not as its presence/absence is not an immanent property of an LLM. Also there may be use cases when we have LoRA, but don't use KV cache optimizations by some reason, but it's still an LLM like model. In that case we would call decomposition while shouldn't, which would end up in perf degradations of certain models.

Reply:
The solution with property gives a way to control it depending on the model architecture that is possible in GenAI.